FEAT: Improve documentation for annotation cache (#1248)
nf-core/sarek: docs/usage.md

Since both Snpeff and VEP are internally figuring the path towards the specific cache version / species, `annotation-cache` is using an extra set of keys to specify the species and genome build.

> So if you are using this resource, please either set `--use_annotation_cache` to use the AWS annotation cache, or point towards your own cache folder structure matching the expected structure.

Author: ```suggestion
So if you are using this resource, please either set `--use_annotation_cache_keys` to use the AWS annotation cache, or point towards your own cache folder structure matching the expected structure.
```

Reviewer: I actually find it without "Keys" more obvious, but maybe too big of a change at this point

> So if you are using this resource, please either set `--use_annotation_cache` to use the AWS annotation cache, or point towards the specific species, genome and build matches the directory structure within your own cache folder structure.

Author: ```suggestion
So if you are using this resource, please either set `--use_annotation_cache` to use the AWS annotation cache, or point towards your own cache folder structure matching the expected structure.
```

> So if you are using this resource, please either use the `--use_annotation_cache_keys`, or point towards the specific species, genome and build matches the directory structure within the cache.

Reviewer: I am stillam still mildly fussy on this, but from what i understand now: `--use_annotation_cache_keys` is the equivalent to `igenomes_base`. Would i make sense to give it a similar name: `--annotation_cache_base` and set it b default to the AWS ressource, same as igenomes?

Author: not really actually.
`vep_cache` and `snpeff_cache` would be equivalent to `igenomes_base`.
`use_annotation_cache_keys` is just a boolean so that Sarek knows there is an extra layer in the organisation of the cache because of cloud related logic.

Reviewer: ```suggestion
So if you are using this resource, please either set `--use_annotation_cache` to use the AWS annotation cache, or point towards the specific species, genome and build matches the directory structure within your own cache folder structure.
```
